plugins/todo-backend/src/plugin.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createBackendPlugin,
+  coreServices,
+} from '@backstage/backend-plugin-api';
+import { CompoundEntityRef, parseEntityRef } from '@backstage/catalog-model';
+import { InputError } from '@backstage/errors';
+import express, { Router } from 'express';
```

Comment:
This is a bit of a gotcha - you should absolutely avoid constructing `express` routers, you should use `'express-promise-router` like [other plugins do](https://github.com/backstage/backstage/blob/b25b4bda4e8af306732bba92ec22e986b28aad27/plugins/catalog-backend/src/service/createRouter.ts#L28). Otherwise, async errors will crash badly instead of getting properly caught as part of the request handler.